gui/web/src/components/atoms/Button/Button.js

```diff
@@ -34,8 +45,34 @@ class Button extends Component {
     variant: PropTypes.string,
     onClick: PropTypes.func,
     loading: PropTypes.bool,
-    disabled: PropTypes.bool
+    disabled: PropTypes.bool,
+    trackEvent: PropTypes.func,
+    eventName: PropTypes.string,
   };
+  
+  sendMetricEvent() {
+    if (this._asyncRequests["sendMetricEvent"]) {
+      return
+    }
+    
```

Comment:
we want to throw an error here if `eventName` is undefined or empty -- it should be defined for everyone and should not be empty.

Reply:
maybe the right thing to do is to let React handle part of this for us.
i.e. the `eventName` prop should be `PropTypes.string.isRequired` and then we only need to check if this is an empty string here. if it's an empty string we should console.log an error before returning.

if React can handle the validation for us as well, something like this (made up my syntax here):
`PropTypes.string.isRequired(exclude=[""])` then that would be great. IIRC React has some validation functionality like this, but don't spend too much time on trying to figure it out if it's not straightforward.

Reply:
handled in new PR I'm putting together